lib/components/steps/mapping/MappingTab.tsx

```diff
@@ -180,18 +186,26 @@ const MappingTab = ({defaultCollection}: MappingProps) => {
         }
     }
 
-
-    const handleFiltering = useCallback((searchTerm: string) => {
+    const handleFiltering = useCallback((searchTerm: string, checked: CheckedState) => {
         const filteredData = Object.keys(datasetMapping).filter(variableName => {
             const variableNameMatch = variableName.toLowerCase().includes(searchTerm.toLowerCase());
-            const preciseAbbreviation = datasetMapping[variableName][headerIndexes.preciseAbbreviation] || '';
+            const row = datasetMapping[variableName]
+            const preciseAbbreviation = row[headerIndexes.preciseAbbreviation] || '';
             const preciseAbbreviationMatch = preciseAbbreviation.toLowerCase().includes(searchTerm.toLowerCase());
 
+            const entityType = getType(row, headerIndexes);
+            const isAnyTrue = Object.values(checked).some(value => value === true);
+    
+            if (isAnyTrue) {
+                return checked[entityType];
```

Comment:
Note (non-blocking): This assumes that all filters will be on the row type. Ideally I would prefer a more generic approach but I won't mark it as a blocker since the code correctly addresses the current scope.